src/ray/core_worker/task_manager.h

```diff
@@ -632,27 +632,18 @@ class TaskManager : public TaskFinisherInterface, public TaskResubmissionInterfa
 
     void SetStatus(rpc::TaskStatus new_status) {
       auto new_tuple = std::make_tuple(spec.GetName(), new_status, is_retry_);
-      counter.Swap(status, new_tuple);
+      if (IsPending()) {
+        counter.Swap(status, new_tuple);
+      } else {
+        // FINISHED and FAILED are monotonically increasing.
+        // TODO(jjyao): We should use Counter instead of Gauge
```

Comment:
Curious question: IIUC, the task status number is already tracked by a counter map. So here "use Counter" do you mean we should use a data structure that can only be monotonically increased instead of being able to both increase & decrease?

Reply:
Yea. The current `CounterMap` is for GAUGE only.